Quantum-Control-Applications/ESR/configuration.py

```diff
@@ -0,0 +1,537 @@
+    gauss = (amp*gaussian(gauss_length, sigma)).tolist()
+    return gauss[:gauss_length//2] + [amp] * flat_top
+
+
+#############
+# VARIABLES #
+#############
+
+qop_ip = "127.0.0.1"
+qop_port = 80
+
+# Frequencies
+resonator_if = -30e6  # in Hz
+
+# ensemble_if = -30e6  # in Hz
+ensemble_if = 0e6  # in Hz
+mixer_ensemble_g = 0.0
+mixer_ensemble_phi = 0.0
+# ensemble_if = 37e6  # in Hz
+
+# LOs are used in plots. They can also be used marking mixer elements in the config.
+# On top of that they can also be used for setting LO sources (this make sure that everything is in sync)
+ensemble_LO = 2.8e9  # in Hz
+
+# Readout parameters
+const_amp = 0.1  # in V
+const_len = 60  # in ns
+
+short_readout_len = 80  # in ns
+short_readout_amp = 0.4  # in V
+
+readout_len = 400  # in ns
+readout_amp = 0.4  # in V
+
+long_readout_len = 70000  # in ns
+long_readout_amp = 0.1  # in V
+
+time_of_flight = 160  # Time it takes the pulses to go through the RF chain, including the device.
+
+# Qubit parameters:
+saturation_amp = 0.2  # in V
+saturation_len = 50000  # Needs to be several T1 so that the final state is an equal population of |0> and |1>
+
+# Pi pulse parameters
+pi_len = 48  # in units of ns
+pi_amp = 0.3  # in units of volts
+pi_wf = (pi_amp * (gaussian(pi_len, pi_len / 5) -
+                   gaussian(pi_len, pi_len / 5)[-1])).tolist()  # waveform
+minus_pi_wf = ((-1) * pi_amp * (gaussian(pi_len, pi_len / 5) -
+                   gaussian(pi_len, pi_len / 5)[-1])).tolist()  # waveform
+
+# Pi_half pulse parameters
+pi_half_len = int(pi_len/2)  # in units of ns
+pi_half_amp = pi_amp  # in units of volts
+pi_half_wf = (pi_half_amp * (gaussian(pi_half_len, pi_half_len / 5) -
+                             gaussian(pi_half_len, pi_half_len / 5)[-1])).tolist()  # waveform
+minus_pi_half_wf = ((-1) * pi_half_amp * (gaussian(pi_half_len, pi_half_len / 5) -
+                             gaussian(pi_half_len, pi_half_len / 5)[-1])).tolist()  # waveform
+
+# Subtracted Gaussian pulse parameters
+gauss_amp = 0.3  # The gaussian is used when calibrating pi and pi_half pulses
+gauss_len = 20  # The gaussian is used when calibrating pi and pi_half pulses
+gauss_wf = (gauss_amp * (gaussian(gauss_len, gauss_len / 5) -
+                         gaussian(gauss_len, gauss_len / 5)[-1])).tolist()  # waveform
+
+# Note: a subtracted Gaussian pulse has a more narrow spectral density than a regular gaussian
+# it becomes useful in short pulses to reduce leakage to higher energy states
+
+# Flux:
+square_flux_amp = 0.3
+minus_square_flux_amp = -0.3
+triangle_flux_amp = 0.3
+triangle_wf = [triangle_flux_amp * i/7 for i in range(8)] + [triangle_flux_amp * (1 - i/7) for i in range(8)]
+
+initialization_len = 1000  # in ns
+activation_len = 500  # in ns
+
+config = {
+
+    'version': 1,
+
+    'controllers': {
+
+        'con1': {
+
+            'type': 'opx1',
+
+            'analog_outputs': {
+                1: {'offset': +0.0},  # qubit I
```

Comment:
the comments here are wrong, either update or remove

Reply:
cool